monai/utils/misc.py

```diff
@@ -821,3 +823,29 @@ def check_kwargs_exist_in_class_init(cls, kwargs):
     extra_kwargs = input_kwargs - init_params
 
     return extra_kwargs == set(), extra_kwargs
+
+
+def run_cmd(cmd_list: list[str], **kwargs: Any) -> subprocess.CompletedProcess:
+    """
+    Run a command by using ``subprocess.run`` with capture_output=True and stderr=subprocess.STDOUT
+    so that the raise exception will have that information. The argument `capture_output` can be set explicitly
+    if desired, but will be overriden with the debug status from the variable.
+
+    Args:
+        cmd_list: a list of strings describing the command to run.
+        kwargs: keyword arguments supported by the ``subprocess.run`` method.
+
+    Returns:
+        a CompletedProcess instance after the command completes.
+    """
+    debug = MONAIEnvVars.debug()
+    kwargs["capture_output"] = kwargs.get("capture_output", debug)
+
+    try:
+        return subprocess.run(cmd_list, **kwargs)
+    except subprocess.CalledProcessError as e:
+        if not debug:
+            raise
+        output = repr(e.stdout).replace("\\n", "\n").replace("\\t", "\t")
+        errors = repr(e.stderr).replace("\\n", "\n").replace("\\t", "\t")
```

Comment:
Was there a specific reason to use this pattern rather than `outputs  = str(e.stdout)`? If stdout and stderr aren't going to contain much text this would be fine but if they are large blocks of output we might want to use `textwrap.indent` to format things a little.

Reply:
The stdout/stderr is in bytes because binary data could actually be written but also weird errors can cause invalid characters which one would want to see. An alternative is something like `e.stdout.decode(errors="replace")` which I feel is a more reliable way of decoding a bytes into a string and will sort all escapes and not just \n and \t.

Reply:
Thank you @ericspod for the suggestion. I will try it and update the PR

Reply:
Fixed in [09e55da](https://github.com/Project-MONAI/MONAI/pull/6596/commits/09e55da44239a95c91b656894c0a6db5be9fb6ee)

Reply:
Hi @ericspod , I will mark this is resolved if there is no more comments. Thanks!